vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueType
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueType Kind;
+    UInt64 Value;
+  };
+
+  [default_interface]
+  runtimeclass JsiPreparedJavaScript
+  {
+  };
+
+  delegate JsiValueData JsiHostFunction(IJsiRuntime runtime, JsiValueData thisValue, JsiValueData[] args);
+
+  [webhosthidden]
+  interface IJsiHostObject
+  {
+    JsiValueData GetProperty(IJsiRuntime runtime, UInt64 name);
+    void SetProperty(IJsiRuntime runtime, UInt64 name, JsiValueData value);
+    IVectorView<UInt64> GetPropertyNames(IJsiRuntime runtime);
+  };
+
+  [webhosthidden]
+  interface IJsiRuntime
+  {
+    JsiValueData EvaluateJavaScript(IJsiBuffer buffer, String sourceUrl);
+    JsiPreparedJavaScript PrepareJavaScript(IJsiBuffer buffer, String sourceUrl);
+    JsiValueData EvaluatePreparedJavaScript(JsiPreparedJavaScript js);
+    UInt64 Global { get; };
+    String Description { get; };
+    Boolean IsInspectable { get; };
+
+    UInt64 CloneSymbol(UInt64 symbol);
+    UInt64 CloneString(UInt64 str);
+    UInt64 CloneObject(UInt64 obj);
+    UInt64 ClonePropertyNameId(UInt64 propertyNameId);
+
+    UInt64 CreatePropertyNameIdFromAscii(UInt8[] ascii);
+    UInt64 CreatePropertyNameIdFromUtf8(UInt8[] utf8);
+    UInt64 CreatePropertyNameIdFromString(UInt64 str);
+    void PropertyNameIdToUtf8(UInt64 propertyNameId, JsiDataHandler utf8Handler);
+    Boolean ComparePropertyNameIds(UInt64 left, UInt64 right);
```

Comment:
What about ArePropertyNameIdsEqual? By using compare it feels like you can sort them.

Reply:
I changed it to PropertyNameIdEquals - the name seems to be more align with names used in .Net and Windows SDK frameworks.

---
In reply to: [451858018](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r451858018) [](ancestors = 451858018)